File: frontend/src/lib/api/market.ts
import { apiClient } from "@/lib/apiClient";
import type { SkillDemandOut } from "@/lib/api/skills";

export interface MarketInsightsOut {
  top_skills: SkillDemandOut[];
  trend_bullets: string[];
  summary: string;
}

export const marketApi = {
  insights: () => apiClient.get<MarketInsightsOut>("/market/insights"),
};

File: backend/app/schemas/market.py
"""Pydantic schemas for `GET /market/insights` (B7)."""
from __future__ import annotations

from pydantic import BaseModel, Field

from app.schemas.skill import SkillDemandOut


class MarketInsightsOut(BaseModel):
    top_skills: list[SkillDemandOut] = Field(default_factory=list)
    trend_bullets: list[str] = Field(default_factory=list)
    summary: str
